model.py

```diff
@@ -9928,4 +9939,84 @@ def numericrange_to_tuple(r):
 def tuple_to_numericrange(t):
     """Helper method to convert a tuple to an inclusive NumericRange."""
     return NumericRange(t[0], t[1], '[]')
+
+def site_configuration_has_changed(_db, timeout=1):
+    """Call this whenever you want to indicate that the site configuration
+    has changed and needs to be reloaded.
+
+    This is automatically triggered on relevant changes to the data
+    model, but you also should call it whenever you change an aspect
+    of what you consider "site configuration", just to be safe.
+
+    :param _db: Either a Session or (to save time in a common case) an
+    object that can be associated with or turned into a Session.
+
+    :param timeout: Nothing will happen if it's been fewer than this
+    number of seconds since the last site configuration change was
+    recorded.
+
+    """
+    now = datetime.datetime.utcnow()
+    last_update = Configuration.site_configuration_last_update()
+    if not last_update or (now - last_update).total_seconds() > timeout:
+        # The configuration last changed more than `timeout` ago, which
+        # means it's time to reset the Timestamp that says when the
+        # configuration last changed.
+        needs_commit = False
+        # Convert something that might not be a Connection object into
+        # a Connection object.
+        if isinstance(_db, Base):
+            _db = Session.object_session(_db)
+        elif isinstance(_db, Connection):
+            _db = Session(_db)
+            needs_commit = True
```

Comment:
I'm not sure about this because it's one of the random things I tried when getting something to work, but unlike many changes of that sort I think it holds up.